pkg/reconciler/autoscaling/hpa/keda_hpa.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hpa
+
+import (
+	"context"
+	"fmt"
+
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	autoscalingv2listers "k8s.io/client-go/listers/autoscaling/v2"
+
+	nv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	autoscalingv1alpha1 "knative.dev/serving/pkg/apis/autoscaling/v1alpha1"
+	"knative.dev/serving/pkg/autoscaler/config/autoscalerconfig"
+	pareconciler "knative.dev/serving/pkg/client/injection/reconciler/autoscaling/v1alpha1/podautoscaler"
+	areconciler "knative.dev/serving/pkg/reconciler/autoscaling"
+	"knative.dev/serving/pkg/reconciler/autoscaling/config"
+
+	"knative.dev/autoscaler-keda/pkg/client/clientset/versioned"
+	kedav1alpha1 "knative.dev/autoscaler-keda/pkg/client/listers/keda/v1alpha1"
+	"knative.dev/autoscaler-keda/pkg/reconciler/autoscaling/hpa/resources"
+)
+
+// Reconciler implements the control loop for the HPA resources.
+type Reconciler struct {
+	*areconciler.Base
+
+	kubeClient kubernetes.Interface
+	kedaClient versioned.Interface
+	kedaLister kedav1alpha1.ScaledObjectLister
+	hpaLister  autoscalingv2listers.HorizontalPodAutoscalerLister
+}
+
+// Check that our Reconciler implements pareconciler.Interface
+var _ pareconciler.Interface = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (c *Reconciler) ReconcileKind(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler) pkgreconciler.Event {
+	ctx, cancel := context.WithTimeout(ctx, pkgreconciler.DefaultTimeout)
+	defer cancel()
+
+	logger := logging.FromContext(ctx)
+	logger.Debug("PA exists")
```

Comment:
want to keep that?

Reply:
I copied it from the Knative HPA logic, we can remove.